packages/next/src/server/lib/mock-request.ts

```diff
@@ -135,6 +172,27 @@ export class MockedResponse extends Stream.Writable implements ServerResponse {
     })
   }
 
+  public appendHeader(name: string, value: string | string[]): this {
+    if (Array.isArray(value)) {
+      for (const v of value) {
+        this.headers.append(name, v)
+      }
+    } else {
+      this.headers.append(name, value)
+    }
```

Comment:
Nit pick: you can simplify this

```suggestion
    const values = Array.isArray(value) ? value : [value]
    for (const v of values) {
      this.headers.append(name, v)
    }
```